### PR TITLE
fix: [CDS-54529]: Reconciliation support added for artifact source templates in service studio

### PR DESCRIPTION
**Pull request by samarth-sharma31**
#### Screenshots

https://user-images.githubusercontent.com/96409598/236930521-e1f95c1f-52d6-4673-991b-55b88e5c90bd.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx [hunk 1]
```diff
@@ -189,58 +221,75 @@ function ServiceConfiguration({
...
+            selectedView={selectedView}
+            onChange={nextMode => {
+              handleModeSwitch(nextMode)
             }}
-            isEditModeSupported={!isReadonly}
-            existingJSON={serviceData}
-            bind={setYamlHandler}
-            schema={serviceSchema?.data}
-            height={isServiceEntityModalView ? 540 : 700}
+            //   disableToggle={!inputSetTemplateYaml}
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx:246]: nit: commented code

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx [hunk 2]
```diff
@@ -189,58 +221,75 @@ function ServiceConfiguration({
...
+            />
+            {isReadonly || !isYamlEditable ? (
+              <div className={css.buttonsWrapper}>
+                <Tag>{getString('common.readOnly')}</Tag>
+                <RbacButton
+                  permission={{
+                    resource: {
+                      resourceType: ResourceType.SERVICE,
+                      resourceIdentifier: defaultTo(serviceId, '')
+                    },
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx:279]: Please check if we need 
`resourceScope: {
                  accountIdentifier: accountId,
                  orgIdentifier,
                  projectIdentifier
                },`
                
Considering we have service at acct and org also.

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx [hunk 3]
```diff
@@ -100,6 +119,68 @@ function ServiceStudioDetails(props: ServiceStudioDetailsProps): React.ReactElem
...
+    try {
+      const response = await validateTemplateInputsPromise({
+        queryParams: {
+          accountIdentifier: accountId,
+          orgIdentifier,
+          projectIdentifier,
+          identifier: serviceId
+        }
+      })
+      clear()
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:132]: can we have this clear() outside the try-catch block? thoughts? 🤔

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx [hunk 4]
```diff
@@ -100,6 +119,68 @@ function ServiceStudioDetails(props: ServiceStudioDetailsProps): React.ReactElem
...
+
+        showSuccess(getString('common.serviceUpdated'))
+        setShouldShowOutOfSyncError(false)
+        fetchPipeline({ forceFetch: true, forceUpdate: true })
+        setIsDeploymentTypeDisabled?.(!!reconciledService.service?.serviceDefinition?.type)
+      } else {
+        throw response
+      }
+    } catch (e: any) {
+      showError(e?.data?.message || e?.message || getString('commonError'))
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:178]:
```suggestion
      showError(getErrorInfoFromErrorObject(e))
```
Please check if this satisfy the requirement

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx [hunk 5]
```diff
@@ -247,6 +334,23 @@ function ServiceStudioDetails(props: ServiceStudioDetailsProps): React.ReactElem
...
+            <Popover className={Classes.DARK} position={Position.LEFT}>
+              <Button variation={ButtonVariation.ICON} icon="Options" aria-label="service studio menu actions" />
+              <Menu style={{ backgroundColor: 'unset' }}>
+                <RbacMenuItem
+                  icon="refresh"
+                  text={getString('pipeline.outOfSyncErrorStrip.reconcile')}
+                  disabled={isReadonly}
+                  onClick={handleReconcileClick}
+                  permission={{
+                    permission: PermissionIdentifier.EDIT_SERVICE,
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:346]: similar as above.

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx:246]: This is not new code - actually this is existing piece, we have just moved it around.
I'll request to owner to take a look at this and maybe remove it in a followup PR

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx (hunk 2 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx:279]: sure, checking

Comment (samarth-sharma31) [reply src/modules/75-cd/components/Services/ServiceStudio/ServiceConfiguration/ServiceConfiguration.tsx:279]: I don't think resourceScope will be required here this is existing piece of code which we have just moved - earlier it was present in lines 226 - 238

@AshwinBhatkal-Harness can you please confirm if resourceScope is required, we can add it in a followup pr

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx (hunk 3 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx]: done

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx (hunk 4 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx]: done

File: src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx (hunk 5 above)
Comment (samarth-sharma31) [reply src/modules/75-cd/components/Services/ServiceStudio/ServiceStudioDetails.tsx:346]: done

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Comment by samarth-sharma31** [comment]
Coverage check is breaking due to branch coverage for files -> 
Screenshot 2023-05-11 at 10 31 52 AM

**Review by YounixIO — APPROVED** [review]

**Pull request merged**